packages/react-utilities/src/hooks/useId.ts

```diff
@@ -21,5 +21,11 @@ export function resetIdsForTests(): void {
 export function useId(prefix?: string, providedId?: string): string {
   const contextValue = useSSRContext();
 
-  return React.useMemo(() => providedId || `${prefix}${++contextValue.current}`, [prefix, providedId, contextValue]);
+  return React.useMemo(() => {
+    if (providedId) {
+      return providedId;
+    }
+
+    return prefix ? `${prefix}${++contextValue.current}` : `${++contextValue.current}`;
```

Comment:
definitely a nit, but this is my personal preference, but it's up to you

```suggestion
    return `${prefix ?? ''}${++contextValue.current}` ;
```

Reply:
That's much nicer!